src/cron-job.ts

```diff
@@ -0,0 +1,227 @@
+import { ApiObject, Duration, Lazy, Cron, ApiObjectMetadata } from 'cdk8s';
+import { Construct } from 'constructs';
+import * as k8s from './imports/k8s';
+import { JobProps } from './job';
+import { RestartPolicy } from './pod';
+import * as workload from './workload';
+
+/**
+ * Concurrency policy for CronJobs.
+ */
+export enum ConcurrencyPolicy {
+  /**
+   * This policy allows to run job concurrently.
+   */
+  ALLOW = 'Allow',
+  /**
+   * This policy does not allow to run job concurrently. It does not let a new job to be scheduled if the previous one is not finished yet.
+   */
+  FORBID = 'Forbid',
+  /**
+   * This policy replaces the currently running job if a new job is being scheduled.
+   */
+  REPLACE = 'Replace',
+}
+
+/**
+ * Properties for `CronJob`.
+ */
+export interface CronJobProps extends JobProps {
+  /**
+   * Metadata of the Job that is being scheduled by CronJob.
+   */
+  readonly jobMetadata?: ApiObjectMetadata;
```

Comment:
Adding a custom name here has no effect. CronJob decides the name of the job being scheduled on its own.

Reply:
Hmm. Then lets just remove this for now, I don't see a good use-case for explicitly setting this metadata right now. We can always add if needed.

Reply:
Will remove it.